### DISCUSSION TITLE
Sudden issues with menu options: not sure it is tsml related but asking for input

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#605] [Q&A] [answered]
I am writing this to find out if any of you have seen the same behaviour described below. When you read on, you will see why I am posting here. As the heading says: I have no idea if there is a connection to the tsml plugin, but better safe than sorry.

By accident I found out yesterday that some of my Wordpress menu items were damaged. The original label of the items flipped to 'new meeting' when I checked an existing menu, without making a change. The menu still showed the link to the correct webpage. Just the label was messed up. I was stumped. I ended up finding multiple errors in the new Wordpress widget are, even referring to plugins I don't even have. The only thing I found online to rectify this was to install a plugin 'classic widget'. The issue was largely gone after that, but it does pop up now and then.

I am on WP 5.8.2 and I am on TSML 3.12.1. I have a very straight forward website with no commenting or ping backs. I have limited visuals and most of them are from screenshots, so not very big. I talked to my hosting provider the other day and they said that the size of the website is not even at 50% of what I am allowed with shared hosting. I am not on managed Wordpress, I am on Wordpress with CPanel at GoDaddy.

The reason I am posting this here is to see if others have had this issue as well and how they fixed it. I am curious why the menu item label changed to 'new meeting'. There really is no reason for that to happen and it sort of hints to TSML because that is the only 'meeting' thing I have on the website.

Looking forward to your comments. If any of you have ideas of where to look when this starts happening again, I would be most grateful to hear. I am not a Wordpress specialist. I have been the website coordinator for my area for 2 years now and things have run smoothly in general. This is the first time something happens that I really do not understand. It is kind of scary because I have no clue of where to look to troubleshoot.

I now have my old widget area back, the plugin 'Classic widget' gets rid of the block structure in Wordpress 5.8.2 that was announced with so much fanfare. I have found no information about a fix.

Any help welcome!

Thanks for reading.....

Happy holidays!

Nicky

**Comment by OSAwebsite** [comment]
Here are 2 screenshots. One is a fragment of the stack of errors in the new widget area within the backend of my site, the other is a screenshot of the 'view > developer > JavaScript section from the browser menu. Maybe that will mean something to someone?


![Java_Developer_Toolerrors](https://user-images.githubusercontent.com/59859095/147376298-c87eb6ec-0886-438a-a0e5-9012e5aefa1b.png)
![WidgetArea_errors](https://user-images.githubusercontent.com/59859095/147376299-dd587de4-28f0-4774-a800-7b28e92cb613.png)

**Comment by kiyote33** [comment]
I can confirm this to be a bug in 3.12.1.  I had the exact same issue with the new site I’m building using the latest Astra theme.  I resolved the issue temporarily by rolling back to 3.12.0…

 

Wayne

 

From: OSAwebsite ***@***.***> 
Sent: December 24, 2021 7:42 PM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [code4recovery/12-step-meeting-list] Sudden issues with menu options: not sure it is tsml related but asking for imput (Discussion #605)

 

I am writing this to find out if any of you have seen the same behaviour described below. When you read on, you will see why I am posting here. As the heading says: I have no idea if there is a connection to the tsml plugin, but better safe than sorry.

By accident I found out yesterday that some of my Wordpress menu items were damaged. The original label of the items flipped to 'new meeting' when I checked an existing menu, without making a change. The menu still showed the link to the correct webpage. Just the label was messed up. I was stumped. I ended up finding multiple errors in the new Wordpress widget are, even referring to plugins I don't even have. The only thing I found online to rectify this was to install a plugin 'classic widget'. The issue was largely gone after that, but it does pop up now and then.

I am on WP 5.8.2 and I am on TSML 3.12.1. I have a very straight forward website with no commenting or ping backs. I have limited visuals and most of them are from screenshots, so not very big. I talked to my hosting provider the other day and they said that the size of the website is not even at 50% of what I am allowed with shared hosting. I am not on managed Wordpress, I am on Wordpress with CPanel at GoDaddy.

The reason I am posting this here is to see if others have had this issue as well and how they fixed it. I am curious why the menu item label changed to 'new meeting'. There really is no reason for that to happen and it sort of hints to TSML because that is the only 'meeting' thing I have on the website.

Looking forward to your comments. If any of you have ideas of where to look when this starts happening again, I would be most grateful to hear. I am not a Wordpress specialist. I have been the website coordinator for my area for 2 years now and things have run smoothly in general. This is the first time something happens that I really do not understand. It is kind of scary because I have no clue of where to look to troubleshoot.

I now have my old widget area back, the plugin 'Classic widget' gets rid of the block structure in Wordpress 5.8.2 that was announced with so much fanfare. I have found no information about a fix.

Any help welcome!

Thanks for reading.....

Happy holidays!

Nicky

—
Reply to this email directly, view it on GitHub <https://github.com/code4recovery/12-step-meeting-list/discussions/605> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANT2UMZJWFRMY44EQGKEC2DUSUVPLANCNFSM5KXKZ3EA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Comment by OSAwebsite** [comment]
Wow - that is a big bug then!! How do you roll back, I have never done
that...


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by kiyote33** [comment]
Hi Nicky,

 

If you install WP Rollback from the wordpress plugin directory, it will add a rollback option to the plugin list.

 

Hope that helps…

Wayne

**Comment by OSAwebsite** [comment]
OK, that is the plugin that I found as well but did not dare to use yet.
I installed it and took the plunge and do not see any mishaps like in the
past 2 days.
It would be good to hear from the team when this is fixed. Obviously the
newest version is unusable until this is debugged properly.
It shared the living daylights out of me, I can tell you that! It does make
me nervous that a plugin like TSML, that should have nothing to do at all
with other pages, can mess things up like this.... next time I will start
upgrades on my staging site. Lesson learned...

Thanks for your help,
Nicky


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
From where I stand, I can only see a link to the menus in the sidebar
through the widget that shows upcoming meetings.
In no other place, as far as I know, does the TSML plugin act on other than
the meeting pages. Hope this is of help.
I don't dare going back to the new Wordpress 5.8.2 widget area though, have
you tried to do that yourself?


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by 97eighty** [reply]
@OSAwebsite Hi Nicky, were you able to rollback the plugin? @kiyote33 has the Astra theme installed and is getting a similar error. I've installed Astra and am not seeing the error while using the Widget. Have you tried disabling plugins to see if there is a conflict somewhere?

**Comment by OSAwebsite** [comment]
Yep I did the roll back with that plug-in. I have a very simple theme
(iconic one) and very basic plugins. I found the issue not so much in the
widget area. I found it when accessing a random menu. It only showed ‘new
meeting’ in the menu item labels. Scary stuff.

On Sat, Dec 25, 2021 at 7:24 PM 97eighty ***@***.***> wrote:

> @OSAwebsite <https://github.com/OSAwebsite> Hi Nicky, were you able to
> rollback the plugin? @kiyote33 <https://github.com/kiyote33> has the
> Astra theme installed and is getting a similar error. I've installed Astra
> and am not seeing the error while using the Widget. Have you tried
> disabling plugins to see if there is a conflict somewhere?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/605#discussioncomment-1870860>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOIWBF3NTI5S5HCLHYG6CUTUSZOCXANCNFSM5KXKZ3EA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/605/comments/1870860@
> github.com>
>
-- 


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by 97eighty** [reply]
@OSAwebsite Which menu flipped to show "new meetings"? Was this your public menu seen by users? Were you logged in at the time when you saw the menu items change?

**Comment by OSAwebsite** [comment]
All of them. The main nav menu as well as the additional ones for the
sidebar menus.

On Sat, Dec 25, 2021 at 8:12 PM 97eighty ***@***.***> wrote:

> @OSAwebsite <https://github.com/OSAwebsite> Which menu flipped to show
> "new meetings"? Was this your public menu seen by users? Were you logged in
> at the time when you saw the menu items change?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/605#discussioncomment-1870887>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOIWBFYX6JW3TB5EY5TWIS3USZTXLANCNFSM5KXKZ3EA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/605/comments/1870887@
> github.com>
>
-- 


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by kiyote33** [comment]
@97eighty Looks like the issue is in Save.php when the Save Menu button is pushed.  I see there was a change to that file with the 3.12.1 release...

**Reply by kiyote33** [reply]
Seems like the "Use a default meeting title if left blank" update in the 3.12.1 Release is the culprit...

**Comment by kiyote33** [comment]
@97eighty Commented out the following and the problem in 605 went away:
/*	if (empty($post['post_title'])) {
		$title = empty($_POST['location']) ? __('New Meeting', '12-step-meeting-list') : $_POST['location'];
		$title .= in_array($_POST['day'], ['0', '1', '2', '3', '4', '5', '6']) ? ' ' . $tsml_days[$_POST['day']] : '';
		$title .= empty($_POST['time']) ? '' : ' ' . $_POST['time'];
		$post['post_title'] = $title;
	}
*/

**Reply by 97eighty** [reply]
@kiyote33 I worked on that change a few months back. Obviously its interfering.

**Comment by 97eighty** [comment, marked as answer]
Fix in latest v. 3.12.2

**Comment by OSAwebsite** [comment]
I have tested this in my staging site, with the new widget area still
active and it seems to work from then on. Unfortunately it does not revert
the menu items to their original state, so if anyone had trouble with this
and did not adjust the 'new meeting' text back to what it should be, it
must be 'repaired' manually. I already did that with the use of the
'Classic Widget' plugin but I would like to hear back from some more people
before updating to 12.2. Are we 100% sure that the issue is fixed? Because
when it hits, the impact is big and will require a lot of work for admins.

Thanks for looking at this quickly....


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by kiyote33** [reply]
I can confirm that 3.12.2 fixed the problem described in #605...
